File: Leetcode 0283 Move Zeroes [easy] [array]/Move Zeroes [two-pointers].js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let i = 0, j = 0;
    while (i < nums.length && j < nums.length) {
        if (nums[j] !== 0) {
            let temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
            i++;
            j++;
        } else {
            j++;
        }
    }
};

File: Leetcode 0283 Move Zeroes [easy] [array]/Move Zeroes [two-pointers].java
class Solution {
    public void moveZeroes(int[] nums) {
        int i = 0, j = 0;
        while (i < nums.length && j < nums.length) {
            if (nums[j] != 0) {
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                i++;
                j++;
            } else {
                j++;
            }
        }
    }
}